### PR TITLE
Fix #2531

### LINKED ISSUE #2531: `replay_swizzle` argument should come after `error_on_failure`
https://github.com/csarofeen/pytorch/blob/657080b6c8518b6ef279f0d336ef87001f078ee1/third_party/nvfuser/csrc/transform_replay.cpp#L533-L534

### PR DESCRIPTION
**Pull request by naoyam**
Changed ReplayTransformations to take bool parameters explicitly with set methods. This is to avoid accidentally passing those bool arguments in a wrong order. More verbose, but would be safer

Fixes #2531

### CODE REVIEW

File: third_party/nvfuser/csrc/transform_iter.cpp [hunk 1]
```diff
@@ -174,13 +174,8 @@ void ReplayTransformations::handle(Swizzle2D* swizzle_2d) {
 
 ReplayTransformations::ReplayTransformations(
     const std::vector<IterDomain*>& _target_domain,
```
Comment (zasdfgbnm) [inline third_party/nvfuser/csrc/transform_iter.cpp]: Unrelated question: why are their names starting with `_`? Is this some naming convention?

**Review by zasdfgbnm — APPROVED** [review]
LGTM, thanks for the clean up!

File: third_party/nvfuser/csrc/transform_iter.cpp (hunk 1 above)
Comment (naoyam) [reply third_party/nvfuser/csrc/transform_iter.cpp]: I guess that's written by @csarofeen long time ago. Will change.

**Comment by naoyam** [comment]
@zasdfgbnm I added one more commit (34a79b9), which doesn't change any logic but I just tried to clean the code up. Can you please review it as well?

File: third_party/nvfuser/csrc/transform_iter.h [hunk 2]
```diff
@@ -39,80 +39,98 @@ struct id_int_lt {
 // If error_on_failure = false, replay will replay everything it can, and ignore
 // operations it can't.
 class TORCH_CUDA_CU_API ReplayTransformations : public IterVisitor {
- protected:
-  // NOLINTNEXTLINE(cppcoreguidelines-non-private-member-variables-in-classes)
```
Comment (naoyam) [inline third_party/nvfuser/csrc/transform_iter.h:43]: I didn't like this `NOLINTNEXTLINE` annotation, so I moved them to `private` when easily done. Still some have the annotation, but looks cleaner than before.

File: third_party/nvfuser/csrc/transform_iter.h [hunk 3]
```diff
@@ -39,80 +39,98 @@ struct id_int_lt {
...
+
+  // Indicates if we want to replay swizzle ops on the replayed
+  //  tensor.
+  // The swizzle op will be replayed if true,
+  // The swizzle inputs will be directly forwarded, and therefore skipping
+  //  the swizzle op if false.
+  // Currently this options should always be off but
+  //  later we may have cases in scheduling large fusions where
+  //  this functionality could be useful.
+  // NOLINTNEXTLINE(cppcoreguidelines-non-private-member-variables-in-classes)
```
Comment (zasdfgbnm) [inline third_party/nvfuser/csrc/transform_iter.h]: Is this still needed? It is private now.

Comment (naoyam) [reply third_party/nvfuser/csrc/transform_iter.h]: Deleted

**Comment by naoyam** [comment]
There's one bug, which is fixed in [7c320ee](https://github.com/csarofeen/pytorch/pull/2532/commits/7c320eed8cac28d4899576400142e992ed187aec)

**Comment by zasdfgbnm** [comment]
> 

What is the bug? The code looks equivalent to me, it just move the error checking from ctor to `runReplay`.

**Comment by zasdfgbnm** [comment]
> > 
> 
> What is the bug? The code looks equivalent to me, it just move the error checking from ctor to `runReplay`.

Oh I see, in ctor, the `error_on_failure_` is not set.

**Review by zasdfgbnm — APPROVED** [review]

**Comment by naoyam** [comment]
Thanks. I'll merge it once all tests finish.

**Comment by naoyam** [comment]
All are green. Merging.

**Pull request merged**